### ISSUE TITLE
Query Error when compiling apex class

### ISSUE THREAD

**Issue by ralphcallaway** [joeferraro/MavensMate-SublimeText#491]
When saving a particular class one of my colleagues is getting a query result error, logs below 
`Result: [OPERATION FAILED]: QueryResult instance has no attribute 'records'`

## Version
- Platform: OS X 10.9.5
- Sublime: ST 3, build 3065
- MM Plugin v3.5.4

## Raw Response
`{"body": "QueryResult instance has no attribute 'records'", "stack_trace": "Traceback (most recent call last):\n  File \"mm.request\", line 93, in execute\n  File \"mm.commands.metadata\", line 293, in execute\n  File \"mm.local_store\", line 137, in check_for_conflicts\n  File \"suds.sudsobject\", line 154, in __getitem__\nAttributeError: QueryResult instance has no attribute 'records'\n", "success": false}`

## Logs
- [MM.log](https://gist.github.com/anonymous/32b38d99c3f5e50997b2)

**Comment by MWCampPC** [comment]
I'm having the same issue on multiple files.  The delete and recreate method is too time consuming.  Is there any other fix?

**Comment by kidtsunami** [comment]
Anyone experiencing this have steps to reproduce?

**Comment by MWCampPC** [comment]
@kidtsunami I will try to recreate later this week.  I do know I originally created the classes via developer console and it was still open.  I tried to commit and started getting this error.  Once I got it on one class, I got it on others that had previously worked.

**Comment by kidtsunami** [comment]
Thanks!




So, you:




1. Have MavensMate/SublimeText open with the project

2. Add a class in the Developer Console

3? - You say you tried to commit and started getting this error. If the file is created in the developer console, it needs to get downloaded locally to MavensMate somehow. 

4. Got Errors




Did you Clean Project?

**Comment by MWCampPC** [comment]
@kidtsunami on 3, Yes, I refreshed from server and Yes after the error I did a Clean project and even recreated the SB project under another name.

I think it is an issue in Salesforce because in the Developer Console I was getting strange behavior.  Any error I got showed to be under a different class name, and I occasionally got a class already exist error.  I took the suggestion to delete my class (after copying the code) and recreate and Both Mavens Mate and Dev Console started working.

I'm not sure if Mavens Mate caused the issue, will try to recreate next week.

**Comment by pbergner** [comment]
I still get this error infrequently. Any luck recreating?

**Comment by benjamin-meyers** [comment]
I have not found any solution to the issue.  It randomly occurs.




Thanks,

Ben Meyers

**Comment by vrforce** [comment]
I am experiencing the similar error. 
I happen to see this when I modify the code and try save it to server - this happens random.

The only work around I do - do the changes on the developer console, then refresh the local code.

**Comment by FishOfPrey** [comment]
After getting the 
> [OPERATION FAILED]: QueryResult instance has no attribute 'records'
error in Sublime I tried editing the same class directly in the developer console and got:

> FIELD_INTEGRITY_EXCEPTION  
> Failed to create createContainerMember for containerId=undefined: null is not a valid containerId.

Trying to use the ToolingAPI directly gave an INVALID_FIELD_FOR_INSERT_UPDATE StatusCode with the error 

> 0:Error  INVALID_FIELD_FOR_INSERT_UPDATE - Dependent class is invalid and needs recompilation:
DFB.xyzcontroller: line 4, column 61: Dependent class is invalid and needs recompilation:
DFB.xyzsearchpagebase: line 57, column 46: Dependent class is invalid and needs recompilation:
DFB.xyzErrorHandling: line 177, column 23: sObject type 'User' is not supported.

Editing and saving all these classes didn't reveal any problems.

In the end I used the good old web editor at https://na2.salesforce.com/01p400000000001/e
For some reason this basic editor was able to save the changes to the apex class.

I can now refresh the updated class back into Sublime/MavensMate and see the changes. However, I still get the QueryResult records error when attempting to save the ApexClass from Sublime.

**Comment by aazzola** [comment]
Same errore here. 
Paste and save in Developer Console+"Refresh File From Server" in Sublime worked for me
As mentioned by @vrforce (thanks!)

**Comment by harmoney-danielb** [comment]
After getting the same error again in Sublime I tried using the Metadata API MetadataContainer and create Method with an ApexClassMember. The ContainerAyncRequest came back with the State "Failed". It did have a DeployDetails, but the componentFailures were null, so there was no indication of what the problem was.

**Comment by flexchecks** [comment]
I have the same error, but it's only on one particular class.  I tried refreshing the class, refreshing the Classes folder, refreshing the whole project as well as doing all that after resetting the Metadata Container, and nothing fixes it.  

I am able to save via going to the Salesforce Class record.

**Comment by seanhharrison** [comment]
Like @flexchecks above, I also tried everything I could think of. @jpmonette 's solution above worked for me

**Comment by abd3721** [comment]
+1 for @jpmonette 's workaround above

**Comment by pbergner** [comment]
any news on a fix for this one? the workaround is a pain if there are dependencies on the class

**Comment by avinashnelluri** [comment]
I ran into this issue in the past. I've made the changes in Developer Console and refreshed from server.

Now, this process is not working. I'm getting this error every time only for one sandbox. All other projects (sandboxes) are working fine. I even removed the project (workspace) and recreated. Still it didn't work. is there anything I can do with my sandbox settings?

Anybody facing this issue?

**Comment by jkraybill** [comment]
Just weighing here, I just hit this same error. Tried a number of things, including creating a whole new clean project, resetting the metadata container, etc, but the only one that worked was @jpmonette's fix of deleting the class server-side and then re-creating in Sublime. Luckily the affected class wasn't a dependency of anything else :)

**Comment by darrenatyurgosky** [comment]
Same error, @jpmonette 's fix worked

**Comment by joeferraro** [comment]
Hey guys, FYI this issue is fixed in our upcoming beta. will update this issue once the beta group is opened further

**Comment by TGMarcovecchio** [comment]
Argg! Just ran into this same issue. Any permanent fix now?

**Comment by joeferraro** [comment]
@TGMarcovecchio you're more than welcome to install the v5 beta: https://github.com/joeferraro/MavensMate-SublimeText#install

**Comment by archita6** [comment]
@joeferraro  I am also getting the error QueryResult instance has no attribute 'records in the maven`s mate version 6 . When i upgraded it to version 7 beta the compile keeps on running and it is not saved. I have to close sublime to stop that.

**Comment by ralphcallaway** [comment]
@archita6 did you install the related mavenmate app? see https://github.com/joeferraro/MavensMate-SublimeText#version-7-betas

**Comment by aalarson** [comment]
I too am experiencing this issue. I created the class using MavensMate. The first time I tried to modify the class content, I got this error. I could not find the class through the UI, but when I tried to create the class again through the Developer Console I got a duplicate error. Ditto MM. Can't implement the delete and recreate workaround because there is no file to delete, according to SublimeText.
